now_executors/elastic_indexer/elastic_indexer.py

```
@@ -0,0 +1,205 @@
+            new_doc['_id'] = doc.id
+            for chunk in doc.chunks:
+                if chunk.modality == "text":
+                    new_doc['text'] = chunk.text
+                    new_doc['text_embedding'] = chunk.embedding
+                elif chunk.modality == "image":
+                    new_doc['image_embedding'] = chunk.embedding
+            new_doc["_op_type"] = "index"
+            new_doc["_index"] = self.index_name
+            new_docs.append(new_doc)
+        return new_docs
+
+    def _transform_es_results_to_matches(self, es_results: List[Dict]) -> DocumentArray:
+        """
+        Transform a list of results from Elasticsearch into a matches in the form of a `DocumentArray`.
+
+        :param es_results: List of dictionaries containing results from Elasticsearch querying.
+        :return: `DocumentArray` that holds all matches in the form of `Document`s.
+        """
+        matches = DocumentArray()
+        for result in es_results:
+            d = Document(id=result['_id'])
+            d.scores[self.metric] = NamedScore(value=result['_score'])
+            matches.append(d)
+        return matches
+
+    def _extract_embeddings(self, doc: Document) -> Dict[str, np.ndarray]:
+        """
+        Get embeddings from the document. Currently supports at most two
+        modalities and two embeddings (text and image).
+
+        :param doc: `Document` with chunks of text document and/or image document.
+        :return: Embeddings as values in a dictionary, modality specified in key.
+        """
+        embeddings = {}
+        for c in doc.chunks:
+            if c.modality == 'text':
+                embeddings['text_embedding'] = c.embedding
+            elif c.modality == 'image':
+                embeddings['image_embedding'] = c.embedding
+            else:
+                print("Modality not found")
+                raise
+        if not embeddings:
+            print("No embeddings extracted")
```

should we raise an error here?

---

Good point, I guess we should raise an error if we assume by default there should be at least one embedding, and none are extracted.